core/util.go

```diff
@@ -216,6 +217,10 @@ func IsAnyNilOrZero(vals ...interface{}) bool {
 			if len(v) == 0 {
 				return true
 			}
+		case timestamppb.Timestamp:
+			if v.AsTime().IsZero() {
```

Comment:
I think we need this to actually say:
```suggestion
		case *timestamppb.Timestamp:
			if v == nil || v.AsTime().IsZero() {
```
This is because there's a difference between having _type_ `nil` (which is what the first case above catches), and having _type_ "pointer to a timestamppb" but _value_ `nil`. Check out [this playground](https://go.dev/play/p/WaPDlTk5G2e) which demonstrates the difference.

This should be tested [here](https://github.com/letsencrypt/boulder/blob/ca6314fa48c3d3bdc8414a7030e71f3fed2e9695/core/util_test.go#L112).